rocketpy/simulation/monte_carlo.py

```diff
@@ -0,0 +1,666 @@
+            heading=self.flight._randomize_heading(),
+            initial_solution=self.flight.initial_solution,
+            terminate_on_apogee=self.flight.terminate_on_apogee,
+        )
+
+        self._inputs_dict = dict(
+            item
+            for d in [
+                self.environment.last_rnd_dict,
+                self.rocket.last_rnd_dict,
+                self.flight.last_rnd_dict,
+            ]
+            for item in d.items()
+        )
+
+        # Export inputs and outputs to file
+        self.__export_flight_data(
+            flight=monte_carlo_flight,
+            inputs_dict=self._inputs_dict,
+            input_file=input_file,
+            output_file=output_file,
+        )
+
+        average_time = (process_time() - self.start_cpu_time) / self.iteration_count
+        estimated_time = int(
+            (self.number_of_simulations - self.iteration_count) * average_time
+        )
+        self.__reprint(
+            f"Current iteration: {self.iteration_count:06d} | "
+            f"Average Time per Iteration: {average_time:.3f} s | "
+            f"Estimated time left: {estimated_time} s",
+            end="\r",
+            flush=True,
+        )
+
+    def __close_files(self, input_file, output_file, error_file):
+        """Closes all the files."""
+        input_file.close()
+        output_file.close()
+        error_file.close()
+
+    def __finalize_simulation(self, input_file, output_file, error_file):
+        """Finalizes the simulation, closes the files and prints the results."""
+        final_string = (
+            f"Completed {self.iteration_count} iterations. Total CPU time: "
+            f"{process_time() - self.start_cpu_time:.1f} s. Total wall time: "
+            f"{time() - self.start_time:.1f} s\n"
+        )
+
+        self.__reprint(final_string + "Saving results.", flush=True)
+
+        # close files to guarantee saving
+        self.__close_files(input_file, output_file, error_file)
+
+        # resave the files on self and calculate post simulation attributes
+        self.input_file = f"{self.filename}.inputs.txt"
+        self.output_file = f"{self.filename}.outputs.txt"
+        self.error_file = f"{self.filename}.errors.txt"
+
+        print(f"Results saved to {self._output_file}")
+
+    def __export_flight_data(
+        self,
+        flight,
+        inputs_dict,
+        input_file,
+        output_file,
+    ):
+        """Exports the flight data to the respective files."""
+        # Construct the dict with the results from the flight
+        results = {
+            export_item: getattr(flight, export_item)
+            for export_item in self.export_list
+        }
+
+        # Write flight setting and results to file
+        # TODO: use json.dumps (requires custom JSONEncoder child class)
+        input_file.write(f"{json.dumps(inputs_dict, cls=RocketPyEncoder)}\n")
+        output_file.write(f"{json.dumps(results, cls=RocketPyEncoder)}\n")
```

Comment:
The error you're pointing out makes sense, given the f-string interpolation only became "official" in python12.
I agree with the suggested solution.

I need to read more before suggesting anything regarding the second part of your comment.
I think @MateusStano could have an opinion on this regard, since he is the one who managed the file saving problem.

Reply:
One of the things still missing for this PR was a better structure for the saving procedures. We should aim to change it to whatever it is that makes the simulation as fast as possible. 

I don't really now what that procedure would be, but I'm sure there are plenty of better ways to do it.

From the looks of things, we are leaving that to another PR. But before merging I would guarantee that the current execution speed is reasonable. If it is too exaggerated, I would not be comfortable releasing a version of a Monte Carlo analysis that is just inferior to what is on the notebook

Reply:
Ok, I will work on this file saving problem then.

Reply:
Simulations are really faster now! See PRs #582 and #581
Code profiling has allowed me to verify that saving the output files was not taking too much time as we expected before.

I think we are good with the current implementations. More improvements can be done in the future.